utils/serde_utils/Cargo.toml

```diff
@@ -3,8 +3,8 @@
 # (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
 [package]
-name = "deduplicating_array"
-description = "A serde serialization strategy that uses PartialEq to reduce serialized size."
+name = "serde_utils"
```

Comment:
`icu_serde_utils`?

serde_utils already exists.

Also tbh I think it's better for `deduplicating_array` to be its own crate, and other utils to go elsewhere. We can stick this util in provider/core for now

Reply:
@sffc thoughts?

Reply:
`icu_provider::serde::utils` perhaps?  That's easiest for us but least shareable with the ecosystem.  Otherwise I think separate crates?  `deduplicating_array` is one, and `serde_nested_borrow` could be the other

Reply:
Yeah, let's do the module for now (and keep deduplicating_array untouched), but we can turn it into a crate later if we want.

We can also just make serde_with no_std and revisit later when there's less pressure.